spring-cloud-gcp-pubsub-stream-binder/src/test/java/org/springframework/cloud/gcp/stream/binder/pubsub/PubSubMessageChannelBinderTests.java

```diff
@@ -42,17 +47,26 @@
 	@ClassRule
 	public static PubSubEmulator emulator = new PubSubEmulator();
 
+	@Rule
+	public TestRule failOnAbsentEmulator = new TestRule() {
+		@Override
+		public Statement apply(Statement statement, Description description) {
+			if (emulator.getEmulatorHostPort() == null) {
+				fail("Pub/Sub emulator missing");
+			}
+			return statement;
+		}
+	};
+
 	@BeforeClass
 	public static void enableTests() {
-		assumeTrue("PubSub Binder tests are disabled. Please enable them with -Dit.pubsub or -Dit.emulator",
-				"true".equals(System.getProperty("it.pubsub")) || "true".equals(System.getProperty("it.emulator")));
+		assumeThat(System.getProperty("it.pubsub"))
```

Comment:
Why do we still have this `assumeThat()` here?
Why it doesn't work from the `PubSubEmulator.before()`?

Doesn't look like `PubSubEmulator` is going to help us enough: still too much boilerplate code to copy/paste in each test where I would like to use a Pub/Sub emulator...

Reply:
Mike suggested doing away with skipping tests through failed assumptions, and to use Maven profiles instead. What do you think? That would make both the assume* statements and my new rule unnecessary, so it would reduce the boilerplate.

Reply:
Actually, I think I know what you mean. The PubSubEmulator rule can force all tests to be skipped or to be failed based on the value of it.emulator, right?

Reply:
Yes. That's what I mean. Sorry for my pure English.

As a sample we have in Spring AMQP this one: https://github.com/spring-projects/spring-amqp/blob/master/spring-rabbit-junit/src/main/java/org/springframework/amqp/rabbit/junit/BrokerRunning.java#L82

And here is its usage: https://github.com/spring-projects/spring-amqp/blob/master/spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/ContainerShutDownTests.java#L44

So, our AMQP tests go if broker is available or they are skipped otherwise.